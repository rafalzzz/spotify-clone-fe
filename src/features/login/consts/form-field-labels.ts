import { LoginFormKeys } from '@/login/enums/login-form-keys';

export const LOGIN_FORM_LABELS = {
  [LoginFormKeys.EMAIL_OR_USERNAME]: 'Email address or username',
  [LoginFormKeys.PASSWORD]: 'Password',
  [LoginFormKeys.REMEMBER_ME]: 'Remember me',
};
