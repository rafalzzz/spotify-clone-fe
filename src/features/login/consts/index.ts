export { FORM_FIELDS } from './form-fields';
export { INITIAL_VALUES } from './initial-values';
export { FORM_FIELD_PLACEHOLDERS } from './form-field-placeholders';
export { LOGIN_FORM_LABELS } from './form-field-labels';
