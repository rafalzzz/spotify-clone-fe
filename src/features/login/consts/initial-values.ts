import { ELoginFormKeys } from '../types';

export const INITIAL_VALUES = {
  [ELoginFormKeys.LOGIN]: '',
  [ELoginFormKeys.PASSWORD]: '',
  [ELoginFormKeys.REMEMBER_ME]: false,
};
