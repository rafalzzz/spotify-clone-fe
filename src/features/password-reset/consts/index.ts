export { PASSWORD_RESET_FORM_FIELDS, PASSWORD_RESET_COMPLETE_FORM_FIELDS } from './form-fields';
export {
  PASSWORD_RESET_INITIAL_VALUES,
  PASSWORD_RESET_COMPLETE_INITIAL_VALUES,
} from './initial-values';
export { FORM_FIELD_PLACEHOLDERS } from './form-field-placeholders';
export { FORM_LABELS } from './form-field-labels';
