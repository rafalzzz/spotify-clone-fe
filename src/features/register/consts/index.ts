export { FORM_FIELDS } from './form-fields';
export { INITIAL_VALUES } from './initial-values';
export { GENDER_OPTIONS } from './gender-options';
export { MONTH_OPTIONS } from './month-options';
export { FORM_FIELD_PLACEHOLDERS } from './form-field-placeholders';
export { REGISTER_FORM_LABELS } from './form-field-labels';
