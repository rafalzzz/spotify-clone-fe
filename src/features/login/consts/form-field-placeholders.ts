import { LoginFormKeys } from '@/login/enums/login-form-keys';

import { LOGIN_FORM_LABELS } from './form-field-labels';

export const FORM_FIELD_PLACEHOLDERS = {
  [LoginFormKeys.LOGIN]: LOGIN_FORM_LABELS[LoginFormKeys.LOGIN],
  [LoginFormKeys.PASSWORD]: LOGIN_FORM_LABELS[LoginFormKeys.PASSWORD],
};
