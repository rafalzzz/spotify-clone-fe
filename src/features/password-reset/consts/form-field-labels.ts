import { PasswordResetFormKeys, PasswordResetCompleteFormKeys } from '@/password-reset/enums';

export const FORM_LABELS = {
  [PasswordResetFormKeys.LOGIN]: 'Email address or username',
  [PasswordResetCompleteFormKeys.PASSWORD]: 'New password',
  [PasswordResetCompleteFormKeys.REPEAT_PASSWORD]: 'Repeat new password',
};
