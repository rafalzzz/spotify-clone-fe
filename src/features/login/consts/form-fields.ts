import { passwordValidator } from '@/login/utils/validators/password-validator';

import { emailOrUsernameValidator } from '@/validators/email-or-username-validator';

import { EInputType } from '@/enums/input-type';

import { TCustomFormItem } from '@/types/custom-form-item-props';
import { TNonStandardItemProps } from '@/types/non-standard-form-item-props';

import { ELoginFormKeys } from '../types';

import { FORM_LABELS } from './form-field-labels';
import { FORM_FIELD_PLACEHOLDERS } from './form-field-placeholders';

export const FORM_FIELDS: (TCustomFormItem<ELoginFormKeys> | TNonStandardItemProps)[] = [
  {
    type: EInputType.TEXT,
    key: ELoginFormKeys.LOGIN,
    name: ELoginFormKeys.LOGIN,
    label: FORM_LABELS[ELoginFormKeys.LOGIN],
    rules: [{ validator: emailOrUsernameValidator }],
    inputProps: {
      placeholder: FORM_FIELD_PLACEHOLDERS[ELoginFormKeys.LOGIN],
    },
  },
  {
    type: EInputType.PASSWORD,
    key: ELoginFormKeys.PASSWORD,
    name: ELoginFormKeys.PASSWORD,
    label: FORM_LABELS[ELoginFormKeys.PASSWORD],
    rules: [{ validator: passwordValidator }],
    inputProps: {
      placeholder: FORM_LABELS[ELoginFormKeys.PASSWORD],
    },
  },
  {
    type: EInputType.SWITCH,
    key: ELoginFormKeys.REMEMBER_ME,
    name: ELoginFormKeys.REMEMBER_ME,
    switchProps: {
      label: FORM_LABELS[ELoginFormKeys.REMEMBER_ME],
    },
  },
];
