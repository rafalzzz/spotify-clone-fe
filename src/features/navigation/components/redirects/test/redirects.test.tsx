import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

import { ROUTES } from '@/consts/routes';

import { Redirects } from '..';

const mockPush = jest.fn();

jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}));

const renderRedirects = () => render(<Redirects />);

describe('Redirects', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('render component without error', () => {
    const screen = renderRedirects();
    expect(screen).toMatchSnapshot();
  });

  it('should call push method with "/signup" path when user click redirection-to-registration button', async () => {
    const { queryByTestId } = renderRedirects();
    const redirectToRegistrationButton = queryByTestId('redirection-to-registration');
    expect(redirectToRegistrationButton).toBeInTheDocument();

    await userEvent.click(redirectToRegistrationButton as Element);

    expect(mockPush).toHaveBeenCalledWith(ROUTES.REGISTER_USER);
  });

  it('should call push method with "/login" path when user click redirection-to-login button', async () => {
    const { queryByTestId } = renderRedirects();
    const redirectToLoginButton = queryByTestId('redirection-to-login');
    expect(redirectToLoginButton).toBeInTheDocument();

    await userEvent.click(redirectToLoginButton as Element);

    expect(mockPush).toHaveBeenCalledWith(ROUTES.LOGIN_USER);
  });
});
