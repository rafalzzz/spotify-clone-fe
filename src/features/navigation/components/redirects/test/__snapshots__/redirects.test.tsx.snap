// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redirects render component without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="redirects"
      >
        <li>
          <button
            class="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-round ant-btn-default ant-btn-lg redirects__item sign-up-button"
            data-testid="redirection-to-registration"
            role="button"
            type="button"
          >
            <span>
              Sign up
            </span>
          </button>
        </li>
        <li>
          <button
            class="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-round ant-btn-default ant-btn-lg redirects__item sign-in-button"
            data-testid="redirection-to-login"
            role="button"
            type="button"
          >
            <span>
              Sign in
            </span>
          </button>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="redirects"
    >
      <li>
        <button
          class="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-round ant-btn-default ant-btn-lg redirects__item sign-up-button"
          data-testid="redirection-to-registration"
          role="button"
          type="button"
        >
          <span>
            Sign up
          </span>
        </button>
      </li>
      <li>
        <button
          class="ant-btn css-dev-only-do-not-override-12upa3x ant-btn-round ant-btn-default ant-btn-lg redirects__item sign-in-button"
          data-testid="redirection-to-login"
          role="button"
          type="button"
        >
          <span>
            Sign in
          </span>
        </button>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
